src/main/java/net/sf/jabref/model/database/DuplicationChecker.java

```diff
@@ -16,33 +17,34 @@
     // use a map instead of a set since i need to know how many of each key is in there
     private final Map<String, Integer> allKeys = new HashMap<>();
 
-    //##########################################
-    //  usage:
-    //  isDuplicate=checkForDuplicateKeyAndAdd( null, b.getKey() , issueDuplicateWarning);
-    //############################################
-    // if the newkey already exists and is not the same as oldkey it will give a warning
-    // else it will add the newkey to the to set and remove the oldkey
-    public boolean checkForDuplicateKeyAndAdd(String oldKey, String newKey) {
-        // LOGGER.debug(" checkForDuplicateKeyAndAdd [oldKey = " + oldKey + "] [newKey = " + newKey + "]");
+    /**
+     * Usage:
+     * <br>
+     * isDuplicate=checkForDuplicateKeyAndAdd( null, b.getKey() , issueDuplicateWarning);
+     *
+     * If the newkey already exists and is not the same as oldkey it will give a warning
+     * else it will add the newkey to the to set and remove the oldkey
+     *
+     * @return true, if there is a duplicate key, else false
+     */
+    public boolean checkForDuplicateKeyAndAdd(Optional<String> oldKey, Optional<String> newKey) {
```

Comment:
Independent of Optional or not, wouldn't it be possible to write the code like:

``` java
boolean duplicate = false;
if (!oldKey.equals(newKey)) {  // If keys are not the same
if (!oldKey.isPresent()) { // No old key
duplicate = addKeyToSet(newKey.get());
} else {
 removeKeyFromSet(oldKey.get());  // Get rid of old key
 if (newKey.isPresent()) { // Add new key if any
  duplicate = addKeyToSet(newKey.get());          
 } 
}
}

if (duplicate) {
...
}
```

(Replace with corresponding non-optional calls (`Objects.equals()` for the first call).